### PR TITLE
Fix migrations problems with slug added to new models

### LINKED ISSUE #5592: psycopg2.errors.NotNullViolation: column "slug" contains null values
### What I'm trying to achieve
Upgrade from version 2.9.0 to 2.10.0-rc.1. Running the migrate command successfully.

Result:
```
 Applying product.0111_auto_20191209_0437... OK
  Applying product.0112_auto_20200129_0050...Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: column "slug" contains null values


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 328, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 369, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 231, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/operations/fields.py", line 249, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 564, in alter_field
    self._alter_field(model, old_field, new_field, old_type, new_type,
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/postgresql/schema.py", line 152, in _alter_field
    super()._alter_field(
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 710, in _alter_field
    self.execute(
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 142, in execute
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: column "slug" contains null values
``` 

### Steps to reproduce the problem
1. Running 2.9.0 version with some data included
2. Upgrade the docker container, try to run the migrate command

### What I expected to happen
Run the migrate command successfully. The migration still seems to be very buggy. We already had issue before as seen here: #5391 

### Screenshots

**System information**
Operating system:

### PR DESCRIPTION
**Pull request by Boquete**
I want to merge this change because migrations fails with creating slugs; Fixes #5592

# Impact

* [X] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by Boquete** [comment]
one more thing, in 0112 there's another save for instance missing after line 68. I will add it in about an hour

**Comment by Boquete** [comment]
we should also handle warehouses without a name - as it will fail on it too

**Comment by maarcingebala** [comment]
@Boquete Can you reformat your changes with `black`? It's failing on CI now.

**Comment by Boquete** [comment]
@maarcingebala i'm right now in move and made all changes from mobile phone / github desktop so can't do it right now. Will fix asap after I run it on my laptop, although it might be by end of the day

File: saleor/warehouse/migrations/0003_warehouse_slug.py [hunk 1]
```diff
@@ -17,12 +17,13 @@ def create_unique_slug_for_warehouses(apps, schema_editor):
         first_char = warehouse.name[0].lower()
         if first_char != previous_char:
```
Comment (JrmyDev) [inline saleor/warehouse/migrations/0003_warehouse_slug.py:18]: what if some warehouse has same first letter name ?? (same for ProductType and collection)

Comment (IKarbowiak) [reply saleor/warehouse/migrations/0003_warehouse_slug.py:18]: The idea here is to not fetch all objects of the specific type in one request. So instances are sorted by lower name, and then for every new first letter, we get slug values from objects, which slugs start with this letter.
So if some warehouses have the same first letter, new `slug_values` will not be generated. Previously generated slug data will be used in `generate_unique_slug`.

**Comment by Boquete** [comment]
@maarcingebala  fixed formatting w. black

**Comment by maarcingebala** [comment]
Thanks @Boquete!

**Pull request merged**